src/main/groovy/nextflow/trace/HttpPostObserver.groovy

```diff
@@ -0,0 +1,228 @@
+
+        sendHttpMessage("started")
+    }
+
+    /**
+     * Send an HTTP message when the workflow is completed.
+     */
+    @Override
+    void onFlowComplete() {
+        sendHttpMessage("completed")
+    }
+
+    @Override
+    void onProcessCreate(TaskProcessor process) {
+
+    }
+
+    /**
+     * Send an HTTP message when a process has been submitted
+     * @param handler
+     * @param trace
+     */
+    @Override
+    void onProcessSubmit(TaskHandler handler, TraceRecord trace) {
+        sendHttpMessage("process_submit", trace)
+    }
+
+    /**
+     * Send an HTTP message, when a process has started
+     * @param handler
+     * @param trace
+     */
+    @Override
+    void onProcessStart(TaskHandler handler, TraceRecord trace) {
+        sendHttpMessage("process_start", trace)
+    }
+
+    /**
+     * Send an HTTP message, when a process completed
+     * @param handler
+     * @param trace
+     */
+    @Override
+    void onProcessComplete(TaskHandler handler, TraceRecord trace) {
+        sendHttpMessage("process_complete", trace)
+    }
+
+    @Override
+    void onProcessCached(TaskHandler handler, TraceRecord trace) {
+
+    }
+
+    @Override
+    boolean enableMetrics() {
+        return false
+    }
+
+    /**
+     * Send an HTTP message, when a workflow has failed
+     * @param handler
+     * @param trace
+     */
+    @Override
+    void onFlowError(TaskHandler handler, TraceRecord trace) {
+        sendHttpMessage("error", trace)
+    }
+
+    /**
+     * Little helper method that sends a HTTP POST message as JSON with
+     * the current run status, ISO 8601 UTC timestamp, run name and the TraceRecord
+     * object, if present.
+     * @param runStatus The current run status. One of {'started', 'running', 'error', 'completed'}
+     * @param trace A TraceRecord object that contains current process information
+     * @return A Java string, that contains the HTTP request response
+     */
+    protected String sendHttpMessage(String runStatus, TraceRecord trace = null){
+
+        if (!this.httpClient.getUrl()){
+            return "No message send, url was not specified or formatted wrong."
+        }
+
+        // Set the message info
+        def time = new Date().format("yyyy-MM-dd'T'HH:mm:ss'Z'", TimeZone.getTimeZone("UTC"))
+        def messageJson = JSLURPER.parseText(
+                """{ "runName": "$runName", "runId": "$runId", "runStatus":"$runStatus", "utcTime":"$time" }""")
+
+        // Append the trace object if present
+        if (trace)
+            messageJson["trace"] = JSLURPER.parseText(JsonOutput.toJson(trace.store))
+
+        // The actual HTTP request
+        String response = ""
+        try{
+            httpClient.sendHttpMessage(JsonOutput.toJson(messageJson))
+            response = httpClient.getResponse()
+            logHttpResponse()
+        } catch (Exception e){
+            log.error "Could not execute HTTP request."
+            log.debug e.message, e
```

Comment:
Having twice a log entry for the same exception is not a good pattern. Also maybe here the better thing is just to raise the exception.